index.js

```diff
@@ -1,46 +1,79 @@
 'use strict';
-// const dataModel = require('screwdriver-data-model');
+const Datastore = require('screwdriver-datastore-base');
+const schemas = require('screwdriver-data-schema');
+const vogels = require('vogels');
 
-/**
- * @constructor
- * @method EngineDynamodbStore
- * @param  {Object}            config Base configuration to be merged with Dynamodb config
- */
-function EngineDynamodbStore(config) {
-    this.config = config;
+const DEFAULT_REGION = 'us-west-2';
+const TABLE_SCHEMAS = {
+    builds: schemas.build.base,
+    jobs: schemas.job.base,
+    pipelines: schemas.pipeline.base,
+    platforms: schemas.platform.base,
+    users: schemas.user.base
+};
 
-    this.client = null;
-}
+class Dynamodb extends Datastore {
+    /**
+     * Constructs a Dynamodb object
+     * @param  {Object} [config]         Configuration object
+     * @param  {String} [config.region]  AWS region to operate in
+     */
+    constructor(config) {
+        let region = DEFAULT_REGION;
 
-/**
- * gets a single item from Dynamodb datastore
- * @method get
- * @param  {Number}   id       item id
- * @param  {Function} callback function to call
- */
-EngineDynamodbStore.prototype.get = function get(id, callback) {
-    return callback(null, {});
-};
+        if (config && config.region) {
+            region = config.region;
+        }
 
-/**
- * gets an array of items from Dynamodb datastore
- * @method scanAll
- * @param  {Object}   params       query parameters
- * @param  {Function} callback function to call
- */
-EngineDynamodbStore.prototype.scanAll = function scanAll(params, callback) {
-    return callback(null, []);
-};
+        super();
+        vogels.AWS.config.update({ region });
 
-/**
- * Configures the dynamodb store to work for a specific table
- * @method configure
- * @param  {Object}  config configuration
- */
-EngineDynamodbStore.prototype.configure = function configure() {
-    return this;
-};
+        this.client = {};
+        Object.keys(TABLE_SCHEMAS).forEach((tableName) => {
+            this.client[tableName] = vogels.define(tableName, {
+                hashKey: 'id',
+                schema: TABLE_SCHEMAS[tableName],
+                tableName
+            });
+        });
+    }
 
-module.exports = function createDynamodbStore(config) {
-    return new EngineDynamodbStore(config);
-};
+    /**
+     * Obtain an item from the DynamoDB table by primary key
+     * @param  {Object}   config       Configuration object
+     * @param  {String}   config.table Name of the table to interact with
+     * @param  {Object}   config.id    ID of the entry to fetch
+     * @param  {Function} callback     fn(err, data)
+     *                                 err - Error object
+     *                                 data - data from the table
+     */
+    get(config, callback) {
+        const client = this.client[config.table];
```

Comment:
Also eventually we will move to validating the input, which should return an error when the config objects are not correct